extensions/rvadapter/src/test/java/com/google/android/agera/rvadapter/RepositoryAdapterTest.java

```diff
@@ -201,6 +201,36 @@ public void shouldCallRecycleForOnViewRecycled() {
   }
 
   @Test
+  public void shouldCallRecycleForOnRebindViewHolder() {
+    when(repositoryPresenter.getItemCount(ALTERNATIVE_REPOSITORY_ITEM)).thenReturn(1);
+    repositoryAdapter.getItemCount(); //Trigger a refresh
+
+    repositoryAdapter.startObserving();
+    repository.accept(ALTERNATIVE_REPOSITORY_ITEM);
+    runUiThreadTasksIncludingDelayedTasks();
+    repositoryAdapter.stopObserving();
+    repositoryAdapter.onBindViewHolder(viewHolder, 0);
+    repositoryAdapter.onBindViewHolder(viewHolder, 1);
```

Comment:
A ViewHolder will cross over to another RepositoryPresenter (without RV calling recycle) only when they actually share the same view type. Then it's the RepositoryPresenters' mutual responsibility to take care of view holders originating from one another. I'd say it's coder's own responsibility rather than ours.

To stop sharing view types but keep sharing layouts, just alias the layout using <item type="layout">.